validator/sawtooth_validator/execution/executor.py

```diff
@@ -103,12 +103,21 @@ def _future_done_callback(self, request, result):
         self._in_process_transactions_count.dec()
         req = processor_pb2.TpProcessRequest()
         req.ParseFromString(request)
+        # If raw header bytes were sent then de-serialize to get
+        # transaction family name and version
+        request_header = transaction_pb2.TransactionHeader()
```

Comment:
This should be moved to the else clause, because it's confusing that it's set here and then re-set in the if block but used in the else clause.

Reply:
Agree